apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -72,6 +72,10 @@ defmodule OMG.Watcher.BlockGetter do
     maximum_block_withholding_time_ms = Application.fetch_env!(:omg_watcher, :maximum_block_withholding_time_ms)
     maximum_number_of_unapplied_blocks = Application.fetch_env!(:omg_watcher, :maximum_number_of_unapplied_blocks)
 
+    # TODO rethink posible solutions see issue #724
+    # if we do not wait here, `ExitProcessor.check_validity()` may timeouts,
+    # which causes State and BlockGetter to reboot, fetches entire UTXO set again, and then timeout...
+    _ = :sys.get_status(ExitProcessor, 10 * 60_000)
     exit_processor_initial_results = ExitProcessor.check_validity()
```

Comment:
ugh... this is a rather brute force approach. with a bit of reactivness - pub sub this would be far more decoupled and scalable

Reply:
Is there a time factor involved that doesn't allow you to do this properly?

Reply:
We should't have dependencies here from ExitProcessor.
but that would make it difficult for the code with several messages sent between the genserver.

We need to think more about what dependencies our genserves should have

Reply:
There isn't time factor but I don't think pub-sub is better here.
Bringing more complexity won't help us improve this module in the future. As now I consider it good enough.
Thanks for the comment any way, I appreciate your involvement ❗️